server/chat-commands/moderation.ts

```diff
@@ -36,9 +36,9 @@ export const commands: ChatCommands = {
 		}
 
 		if (!this.can('makeroom')) return false;
+		if (room.auth.has(toID(name))) return this.errorReply(`${name} is already a room owner.`);
 
-
-		room.auth.setGroup(user, '#');
+		room.auth.set(toID(name), '#');
```

Comment:
You use `toID(name)` twice; it should be set to a variable `userid` instead.

Reply:
Or possibly `set` could return whether or not it was already that value; combining the check.

Reply:
if it doesn't return `this`, typescript disagrees, since the base type `set` in `Map` returns the map object

Reply:
Unfortunate.

Reply:
This isn't resolved, you're still using `toID(name)`. The old code even correctly uses `userid`!